src/network/connection_handler_task.cpp

```diff
@@ -0,0 +1,43 @@
+#include "network/connection_handler_task.h"
+#include "network/connection_handle.h"
+#include "network/connection_handle_factory.h"
+
+namespace terrier::network {
+
+ConnectionHandlerTask::ConnectionHandlerTask(const int task_id) : NotifiableTask(task_id) {
+  int fds[2];
+  if (pipe(fds) != 0) {
+    NETWORK_LOG_ERROR("Can't create notify pipe to accept connections");
+    exit(1);
```

Comment:
Is this the right error handling? Are there other options that allow continued operation?